### PR TITLE
Auth collaborators

### PR DESCRIPTION
**Pull request by cgwalters**
Two cleanup patches, then a new feature to auth collaborators.  See commit messages for detail.

This change is [Reviewable](https://reviewable.io/reviews/servo/homu/77)

### CODE REVIEW

**Comment by cgwalters** [comment]
For reference, my testing PR: https://github.com/cgwalters/playground/pull/45

File: homu/main.py [hunk 1]
```diff
@@ -359,13 +390,14 @@ def parse_commands(body, username, repo_cfg, state, my_username, db, states, *,
                 state.add_comment(':v: @{} can now approve this pull request'.format(state.delegate))
 
         elif word == 'delegate-':
+            # TODO: why is this a TRY?
```
Comment (Manishearth) [inline homu/main.py]: It should not be.

Comment (Manishearth) [inline homu/main.py]: Oh. It should be. `delegate-` is rarely used and it doesn't matter who does it.

**Comment by Manishearth** [comment]
I'm a bit unhappy that the check got moved down (so it's easier to miss adding the check when making changes), but that's minor.

**Comment by cgwalters** [comment]
Uh, that sounds wacky.  So it only worked for me because my homu was a collaborator or something?  Can we track this as a new issue rather than the comment on closed PR?

**Comment by Manishearth** [comment]
Already filed an issue, and fixed it :)

**Comment by Manishearth** [comment]
(fix is in the currently-open PR for tree closure)

**Pull request merged**